### PR TITLE
UCP: Migrate `DataForAnalyzeStatus` from package `infoschema` to `executor`

### PR DESCRIPTION
**Pull request by yzwqf**
### What problem does this PR solve? 
UCP [#15021](https://github.com/pingcap/tidb/issues/15021)

### What is changed and how it works?
Migrate DataForAnalyzeStatus from package infoschema to executor.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by reafans** [comment]
Hi, @yzwqf, thanks for you contribution, please bound the `DataForAnalyzeStatus` to `memtableRetriever` as a method of it, like `func (e *memtableRetriever) setDataForAnalyzeStatus(ctx sessionctx.Context) `

**Comment by yzwqf** [comment]
> Hi, @yzwqf, thanks for you contribution, please bound the `DataForAnalyzeStatus` to `memtableRetriever` as a method of it, like `func (e *memtableRetriever) setDataForAnalyzeStatus(ctx sessionctx.Context) `
DataForAnalyzeStatus was used by ShowExec.fetchShowAnalyzeStatus, it seems much work to do if so, really?

**Comment by yzwqf** [comment]
> Hi, @yzwqf, thanks for you contribution, please bound the `DataForAnalyzeStatus` to `memtableRetriever` as a method of it, like `func (e *memtableRetriever) setDataForAnalyzeStatus(ctx sessionctx.Context) `

hi, what about this? it seems unitest failed for another reason, I can't helping solving next problem⊙ω⊙

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -1084,3 +1086,35 @@ func (e *memtableRetriever) setDataFromSessionVar(ctx sessionctx.Context) error
 	e.rows = rows
 	return nil
 }
+
+// dataForAnalyzeStatusHelper is a helper function which can be used in show_stats.go
+func dataForAnalyzeStatusHelper(sctx sessionctx.Context) (rows [][]types.Datum) {
```
Comment (zz-jason) [inline executor/infoschema_reader.go]: this function is only called by `setDataForAnalyzeStatus()`, it's not necessary to extract it out of `setDataForAnalyzeStatus()`

**Pull request merged**